### ISSUE TITLE
Skills missing from Change Ship skills menu.

### ISSUE THREAD

**Issue by Frost212** [pyfa-org/Pyfa#1192]
## Bug Report
Racial/T2 Hull skills missing from some "Change Ship skills" Right click menus.
https://puu.sh/w0DbS/7ac719608b.png

### Expected behavior:
The Racial/T2 Hull skill (I.e Amarr frigate, and Assault Frigate) should both show up when you open the "Change ship skills" section of the right click menu while modifying a fit.

### Actual behavior:
The Racial/t2 Hull skills only show up on some ship hulls in the right click menu. (Some are missing one, some are missing both)

### Detailed steps to reproduce:
Open a fit up, right click on a module or on a category header and go to "Change Ship skills". Look for ~insert racial Frigate/cruiser etc~ and/or ~insert Hac/Hic/Ceptor etc~. 

### Fits involved in EFT format (Edit > To Clipboard > EFT):
N/a

### Release or development git branch? Please note the release version or commit hash:
Release Version 1.29.2

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):
Windows 10 x64

### Other relevant information:
https://docs.google.com/spreadsheets/d/1npqrWEMP_6b6px1sGsy973726Lj67qTUaR3oNn-bfZY/edit?usp=sharing

I went through pretty much every hull (Except for Special edition ships >.>) and checked what hulls were missing what. I have made the above spreadsheet with sub-sheet for each size of hull and seperated them by Sub-type (I.e Assault frig, Interceptor, Logistics Frigate)

**Comment by blitzmann** [comment]
@Frost212 Damn, awesome work with the spreadsheet. Excellent bug report :D

IIRC, this is a known limitation of how this thing works. It's been a long while since I looked at the "affecting skills" bit, but I do know it's not 100% accurate for all cases (due to technical limitations). For hull skills, you should look on the modules that the skill is affecting, rather than the hull itself. I know this is counter-intuitive to a degree, but with the way effects work, the skills don't actually change anything on the hull itself, rather they change things on their modules, which is why they don't register with the hull itself.

For example (and please note I don't have the ability to check on any of this at the moment), if Caldari Frigate gives 10% boost to rocket damage per level... that probably won't show up on the "Change Ship Skills". It wouldn't even show up on the Rocket Launcher. Instead, it would show up on the charge itself, since it's the charge's damage that it's affecting.

Whether or not that's what's going on, I won't be sure until I can get to a computer. If that's the case, I could probably add specific skills to the ship listing (for hull bonuses) - it's something I've been thinking about doing for a while now, never looked into.

**Comment by Frost212** [comment]
Just took a quick peak.  You are correct in that they are showing up on the Modules themselves (Checked the phantasm for example, the Caldari cruiser skill ONLY shows up on an afterburner, and the Amarr cruiser shows up on the guns). It would be nice to have them listed with the rest of the generic hull skills (Since it's technically still a ship skill) But as long as it shows up somewhere (Other than the character editor) than i guess that's all that can realistically be expected. Thanks for getting back to me about it quickly though!

**Comment by blitzmann** [comment]
I agree, it would be preferable to have them located on the ship, as that's where the majority of folks would look for them. I have an idea on how to do this that I'll try to test out tonight, but can't promise anything. :)

**Comment by Frost212** [comment]
@blitzmann I found a  similar bug when fitting up a sacrilege (Although this one appears to ACTUALLY be a bug).  The skill "Warhead upgrades" (Which, on the other three ships i checked, shows up ((Gila, Cerb, Damnation)) doesn't show up under "Charge skills".  Wasn't sure if i should open a new issue ticket or not, since it's somewhat the same as the original report i made. (Also of note, currently using Version 1.29.3 Git for one of the bug fixes you put out instead of 1.29.2)

**Comment by blitzmann** [comment]
@Frost212 So, I believe there is another issue at play here. On the sacrilege, check the character that is assigned to the ship. Does it have Warhead Upgrades at "Not Learned"?

IIRC, this is an old bug in which "Not learned" skills don't show up. This is because they aren't applied _at all_ during the fitting calculations, thus the modules and whatnot never know about them. If you switch it to a character that has the skill learned (even All 0), it should show up (which is probably why it's showing up with the other ships you tried it with). Could you confirm this behavior?

**Comment by Frost212** [comment]
@blitzmann Hmm, it does seem you are correct. On my main it does show up in the list (She has it trained to 5, the Alt has it injected, but not trained yet).
